cwms-data-api/src/main/java/cwms/cda/data/dao/AuthDao.java

```diff
@@ -69,28 +70,31 @@ public class AuthDao extends Dao<DataApiPrincipal>{
     public static final String GET_SINGLE_KEY = "select userid,key_name,created,expires from cwms_20.at_api_keys where UPPER(userid) = UPPER(?) and key_name = ?";
     public static final String ONLY_OWN_KEY_MESSAGE = "You may not create API keys for any user other than your own.";
 
-    private boolean hasCwmsEnvMultiOfficeAuthFix;
-    private String connectionUser;
-    private String defaultOffice;
+    private static boolean hasCwmsEnvMultiOfficeAuthFix = false;
+    private static String connectionUser = null;
+    private static String defaultOffice = null;
 
     public AuthDao(DSLContext dsl, String defaultOffice) {
         super(dsl);
         if (getDbVersion() < Dao.CWMS_23_03_16) {
             throw new RuntimeException(SCHEMA_TOO_OLD);
         }
-        this.defaultOffice = defaultOffice;
-        try {
-            connectionUser = dsl.connectionResult(c->c.getMetaData().getUserName());
-            dsl.execute("BEGIN cwms_env.set_session_user_direct(?,?)", connectionUser,defaultOffice);
-            hasCwmsEnvMultiOfficeAuthFix = true;
-        } catch (DataAccessException ex) {
-            if( ex.getLocalizedMessage()
-                  .toLowerCase()
-                  .contains("wrong number or types of arguments in call")) {
-                hasCwmsEnvMultiOfficeAuthFix = false;
+
+        if (AuthDao.defaultOffice == null) {
+            AuthDao.defaultOffice = defaultOffice;
+            try {
+                connectionUser = dsl.connectionResult(c->c.getMetaData().getUserName());
+                dsl.execute("BEGIN cwms_env.set_session_user_direct(?,?)", connectionUser,defaultOffice);
+                hasCwmsEnvMultiOfficeAuthFix = true;
+            } catch (DataAccessException ex) {
+                if( ex.getLocalizedMessage()
+                    .toLowerCase()
+                    .contains("wrong number or types of arguments in call")) {
+                    hasCwmsEnvMultiOfficeAuthFix = false;
+                }
             }
+            AuthDao.RETRIEVE_GROUPS_OF_USER = ResourceHelper.getResourceAsString("/cwms/data/sql/user_groups.sql",this.getClass());
```

Comment:
Is there a reason to do this here vs at the variable declaration?

Reply:
I'm quite sure there was when I initial did the work.

Reply:
Hmm, that's not a bad idea. Let me chew on that one a bit. It is supposed to be usable by all the possible Auth mechanisms.

Reply:
Well, that response was to the comment above but I did go ahead and just do this.